addons/sale/models/res_partner.py

```diff
@@ -64,6 +64,10 @@ def can_edit_vat(self):
             [('partner_id', 'child_of', self.commercial_partner_id.id)]
         )
 
+    def can_edit_info(self):
+        """ this method allows user to change address information. """
+        return not self._has_invoice([('partner_id', '=', self.id)])
+
```

Comment:
Where is this method being used? Did I miss something? :thinking: 
```suggestion
```

Reply:
I have used this method [here](https://github.com/odoo/odoo/pull/166657/files#diff-2acd0e6f67a1190985466e669bc815626745d14981784d828e404f8855af186eR365) to check that partner should not modify some fields when invoice is linked with the address.

Reply:
I think the file `addons/portal/controllers/portal.py` has too much code that's why you are not able to find it because it gets hidden in load diff :sweat_smile: 

here is the code `'has_invoice': partner_sudo.can_edit_info() if partner_sudo else True`,.